### PR TITLE
2849 - SIS2 - Convert internal sample into newCase message

### PR DESCRIPTION
**Pull request by liamtoozer**
# Motivation and Context

We need to change RM so that it can receive a `newCase` message from an external service, instead of RM sending an internal-only sample message.

# What has changed

* Updated config for new topic & subscription and remove sample
* Rename to `newCase`

# How to test?

Run the ATs, make sure everything still passes.

# Links

https://trello.com/c/264gpPpL/2849-sis2-convert-internal-sample-into-newcase-message-5

### CODE REVIEW

**Comment by liamtoozer** [comment]
[SIS2: Convert internal sample into newCase message (5)](https://trello.com/c/264gpPpL/2849-sis2-convert-internal-sample-into-newcase-message-5)

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ Example message:
 {
   "caseId": "b3d63a55-67bd-4f38-a18d-41297d699d5d",
   "collectionExerciseId": "58d971eb-c8f0-45b4-bfd8-af7b9ad8c781",
-  "sample": {
+  "newCase": {
```
Comment (NickGrantONS) [inline README.md:57]: this section should be removed as it's no longer an internal event

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/utils/Constants.java [hunk 2]
```diff
@@ -2,5 +2,5 @@
 
 public class Constants {
   public static final String EVENT_SCHEMA_VERSION = "v0.2_RELEASE";
-  public static final String TOPIC_SAMPLE = "rm-internal-sample";
+  public static final String INBOUND_NEW_CASE_TOPIC = "event_new-case";
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/utils/Constants.java:5]: this constant is only used by tests, so it should live in `TestConstants`

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
Minor tweaks, but otherwise looks great

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/utils/Constants.java (hunk 2 above)
Comment (liamtoozer) [reply src/main/java/uk/gov/ons/ssdc/caseprocessor/utils/Constants.java:5]: It was previously used [here](https://github.com/ONSdigital/ssdc-rm-caseprocessor/blob/57787d7fb7f62fa25af8a19e4e7c5f0ebbdf7993/src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java#L85) but might more sense to just hardcode it?

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java [hunk 3]
```diff
@@ -45,43 +45,45 @@ public SampleReceiver(
...
 
     eventLogger.logCaseEvent(
         newCase,
         OffsetDateTime.now(),
-        "New case created from sample load",
+        "New case created from newCase message",
         EventType.NEW_CASE,
-        createEventDTO(TOPIC_SAMPLE, sample.getJobId(), sample.getOriginatingUser()),
-        RedactHelper.redact(sample),
+        createEventDTO(
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java:84]: no need to create this... use the real one!

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java [hunk 4]
```diff
@@ -45,43 +45,45 @@ public SampleReceiver(
   }
 
   @Transactional
-  @ServiceActivator(inputChannel = "sampleInputChannel", adviceChain = "retryAdvice")
-  public void receiveSample(Message<byte[]> message) {
-    Sample sample = convertJsonBytesToObject(message.getPayload(), Sample.class);
+  @ServiceActivator(inputChannel = "newCaseInputChannel", adviceChain = "retryAdvice")
+  public void receiveNewCase(Message<byte[]> message) {
+    NewCase newCaseMessage = convertJsonBytesToObject(message.getPayload(), NewCase.class);
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java:50]: now it's an event can use the 'proper' handler

Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java:50]:
```suggestion
    EventDTO event = convertJsonBytesToEvent(message.getPayload());
```

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java (hunk 3 above)
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java:84]:
```suggestion
         event.getHeader()
```

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/utils/Constants.java (hunk 2 above)
Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/ssdc/caseprocessor/utils/Constants.java:5]: shouldn't be needed anywhere except integration tests

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java [hunk 5]
```diff
@@ -45,43 +45,45 @@ public SampleReceiver(
...
     newCase = saveNewCaseAndStampCaseRef(newCase);
-    caseService.emitCaseUpdate(newCase, sample.getJobId(), sample.getOriginatingUser());
+    caseService.emitCaseUpdate(
+        newCase, newCaseMessage.getJobId(), newCaseMessage.getOriginatingUser());
 
     eventLogger.logCaseEvent(
         newCase,
         OffsetDateTime.now(),
-        "New case created from sample load",
+        "New case created from newCase message",
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java:82]:
```suggestion
        "New case created",
```

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java [hunk 6]
```diff
@@ -45,43 +46,46 @@ public SampleReceiver(
...
         newCase,
-        OffsetDateTime.now(),
-        "New case created from sample load",
+        event.getHeader().getDateTime(),
+        "New case created",
         EventType.NEW_CASE,
-        createEventDTO(TOPIC_SAMPLE, sample.getJobId(), sample.getOriginatingUser()),
-        RedactHelper.redact(sample),
+        event.getHeader(),
+        RedactHelper.redact(newCasePayload),
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/messaging/NewCaseReceiver.java:88]: no need to call the redact helper... `logCaseEvent` will do that for you already

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
Just that one minor comment about not needing to double up on the RedactHelper

**Review by AdamHawtin — APPROVED** [review]

**Review by NickGrantONS — CHANGES_REQUESTED** [review]
Function `convertJsonBytesToObject` should really move to be a test helper now... it's not used by the prod code

**Review by NickGrantONS — APPROVED** [review]

**Pull request merged**